Jenkinsfile

```diff
@@ -1,92 +1,155 @@
-    script {
-        withCredentials([string(credentialsId: 'PacePython_API_Token',
-                variable: 'api_token')]) {
-          if (isUnix()) {
-            sh """
-                curl -H "Authorization: token ${api_token}" \
-                --request POST \
-                --data '{ \
-                    "state": "${status}", \
-                    "description": "${message} on ${env.JOB_BASE_NAME}", \
-                    "target_url": "$BUILD_URL", \
-                    "context": "${env.JOB_BASE_NAME}" \
-                }' \
-                https://api.github.com/repos/pace-neutrons/pace-python/statuses/${env.GIT_COMMIT}
-            """
-          }
-          else {
-            return powershell(
-            script: """
-                \$body = @"
-                  {
-                    "state": "${status}",
-                    "description": "${message} on ${env.JOB_BASE_NAME}",
-                    "target_url": "$BUILD_URL",
-                    "context": "${env.JOB_BASE_NAME}"
-                  }
-"@
-                [Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"
-                Invoke-RestMethod -URI "https://api.github.com/repos/pace-neutrons/pace-python/statuses/${env.GIT_COMMIT}" \
-                    -Headers @{Authorization = "token ${api_token}"} \
-                    -Method 'POST' \
-                    -ContentType "application/json" \
-                    -Body \$body
-            """,
-            returnStdout: true
-            )
-          }
-        }
-    }
+def name_conda_env(String python_version) {
+  def env_name = "py" + python_version.replace(".","")
+  return env_name
 }
 
+
 pipeline {
 
   agent {
     label get_agent(env.JOB_BASE_NAME)
   }
 
+  environment {
+    ENV_NAME = name_conda_env(env.PYTHON_VERSION)
+  }
+
   stages {
 
+    stage('Notify') {
+      steps {
+        post_github_status("pending", "The build is running")
+      }
+    }
+
+    stage('Display-Environment-Variables') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh 'env'
+          }
+          else {
+            powershell 'Get-ChildItem Env:'
+          }
+        }
+      }
+    }
+
     stage("Build-Pace-Python") {
       steps {
         script {
           if (isUnix()) {
             sh '''
-                podman run -v `pwd`:/mnt localhost/pace_python_builder /mnt/manylinux/jenkins_build_script.sh
+                module purge
+                module load matlab/\$MATLAB_VERSION
+                module load cmake
+                module load conda
+                module load gcc/\$GCC_VERSION
+                conda create -n \$ENV_NAME -c conda-forge python=\$PYTHON_VERSION -y
+                conda activate \$ENV_NAME
+                conda install -c conda-forge setuptools
+                python setup.py bdist_wheel
             '''
-            archiveArtifacts artifacts: 'wheelhouse/*whl'
+            archiveArtifacts artifacts: 'dist/*whl'
```

Comment:
Should this be outside the `if` to avoid duplication?

Reply:
I don't think so in this case. It will only do one each time and when we have multiple configurations running with the different python versions the wheels will be different